In [ ]:
# CultureCompass - Exploratory Data Analysis

## Objective

#This notebook explores the collected cultural event dataset.

#Goals:

#Understand the data quality
#Identify missing values
#Explore event distributions
#Discover patterns across cities and genres
#Generate insights for recommendation modeling

In [6]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt

In [7]:
df = pd.read_parquet("../data/processed/events.parquet")

df.keys()

Index(['event_id', 'event_name', 'genre', 'segment', 'date', 'time', 'city',
       'country', 'venue_id', 'latitude', 'longitude', 'url', 'image_url',
       'event_text'],
      dtype='str')

In [8]:
df["image_url"]

0       https://s1.ticketm.net/dam/a/564/f2895e6a-8172...
1       https://s1.ticketm.net/dam/a/6a5/014ee2b0-0c86...
2       https://s1.ticketm.net/dam/c/df8/81eadad8-4449...
3       https://s1.ticketm.net/dam/a/a9f/81548f92-5ba5...
4       https://s1.ticketm.net/dam/c/ab4/6367448e-7474...
                              ...                        
5134    https://s1.ticketm.net/dam/a/de9/4b5fd57b-0210...
5135    https://s1.ticketm.net/dam/a/3fb/b219bd5d-6fd2...
5136    https://s1.ticketm.net/dam/a/928/84b76085-3684...
5137    https://s1.ticketm.net/dam/a/cfc/61fcb7f2-fcac...
5138    https://s1.ticketm.net/dam/a/cfc/61fcb7f2-fcac...
Name: image_url, Length: 5139, dtype: str

In [10]:
print(df.loc[1, "image_url"])

https://s1.ticketm.net/dam/a/6a5/014ee2b0-0c86-42a9-bdab-ef4b084116a5_SOURCE


In [6]:
print(f"Rows: {df.shape[0]}")
print(f"Columns: {df.shape[1]}")

Rows: 5139
Columns: 14


In [5]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5139 entries, 0 to 5138
Data columns (total 13 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   event_id    5139 non-null   str           
 1   event_name  5139 non-null   str           
 2   genre       5069 non-null   str           
 3   segment     5126 non-null   str           
 4   date        5139 non-null   datetime64[us]
 5   time        5133 non-null   str           
 6   city        5139 non-null   str           
 7   country     5139 non-null   str           
 8   venue_id    5139 non-null   str           
 9   latitude    5042 non-null   float64       
 10  longitude   5042 non-null   float64       
 11  url         5139 non-null   str           
 12  event_text  5139 non-null   str           
dtypes: datetime64[us](1), float64(2), str(10)
memory usage: 1.8 MB


In [7]:
missing = (
    df.isnull()
      .sum()
      .sort_values(ascending=False)
)

missing

longitude     97
latitude      97
genre         70
segment       13
time           6
date           0
event_name     0
event_id       0
city           0
venue_id       0
country        0
url            0
event_text     0
dtype: int64

In [8]:
print(
    "Duplicate event IDs:",
    df["event_id"].duplicated().sum()
)

Duplicate event IDs: 0


In [13]:
city_counts = df["city"].value_counts()
city_counts

city
london       3806
berlin        649
amsterdam     401
vienna        145
london         84
barcelona      54
Name: count, dtype: int64

In [14]:
genre_counts = df["genre"].value_counts()
genre_counts

genre
theatre                    1324
family                     1200
rock                        766
comedy                      568
pop                         362
jazz                        302
miscellaneous                99
alternative                  65
classical                    62
opera                        50
dance/electronic             35
r&b                          29
world                        26
fine art                     21
other                        21
community/civic              20
blues                        15
reggae                       15
undefined                    14
dance                         9
children's theatre            9
hip-hop/rap                   8
multimedia                    6
fairs & festivals             5
variety                       5
cultural                      5
ice shows                     5
latin                         4
soccer                        3
folk                          3
miscellaneous theatre         2
ch

In [15]:
segment_counts = (
    df["segment"]
      .value_counts()
)

segment_counts

segment
arts & theatre    2066
music             1698
miscellaneous     1235
film                67
undefined           56
sports               4
Name: count, dtype: int64

In [16]:
events_per_month = (
    df
    .groupby(df["date"].dt.to_period("M"))
    .size()
)

events_per_month

date
2026-07    1238
2026-08     994
2026-09     571
2026-10     591
2026-11     551
2026-12     401
2027-01     269
2027-02     154
2027-03     146
2027-04      73
2027-05      93
2027-06      37
2027-07       1
2027-09      15
2027-10       1
2027-11       4
Freq: M, dtype: int64

In [17]:
df["text_length"] = (
    df["event_text"]
    .str.split()
    .str.len()
)

df["text_length"].describe()

count    5139.000000
mean        9.675034
std         3.282866
min         3.000000
25%         7.000000
50%        10.000000
75%        11.000000
max        21.000000
Name: text_length, dtype: float64

In [18]:
df[
    [
        "latitude",
        "longitude"
    ]
].describe()

,latitude,longitude
count,5042.000000,5042.000000
mean,51.441813,2.033024
std,1.697641,7.619811
min,0.000000,-77.294899
25%,51.506975,-0.142143
50%,51.513741,-0.129707
75%,51.541020,0.005519
max,52.553280,16.452370


# Imoportant Numbers

- Dataset contains 5139 unique events.

- No duplicate event IDs.

- Most events come from London. 

- The dominant genres are theatre, family,rock, comedy. Music events are less as they divided in subgroups such as rock, pop, jazz, classical, etc. 

- Missing values are minimal. Missing numbers - longitude and latitude:97,  genre:70, segment:13, time:6.

- Date range spans 07-2026 to 11-2027.

These insights suggest the dataset is suitable for building recommendation models.